odoo/orm/domains.py

```diff
@@ -897,6 +951,51 @@ def _optimize_field_search_method(self, model: BaseModel) -> Domain:
             model_label=f"{model.env['ir.model']._get(model._name).name!r} ({model._name})",
         ))
 
+    def _filter_records(self, records):
+        if not records:
+            return lambda _: False
+        if self._opt_level < OptimizationLevel.BASIC:
+            return self.optimize(records, full=False)._filter_records(records)
```

Comment:
What's the point of optimizing each condition independently instead of optimizing the domain as a whole?

Reply:
You can call directly `DomainCondition.filter` which needs to be optimized to avoid handling of all possible inputs. We could optimize the whole domain before filtering, but I saw that in most cases, domains are quite simple (or already optimized) and we can do it lazily.